### PR TITLE
Add Black for autoformatting

### PR DESCRIPTION
**Pull request by cwdavies**
This commit has had `black` (https://github.com/psf/black), the autoformatter, run on the `wagtailflags` package and `setup.py`

It also adds `black --check` to the `tox` `lint` environment and adds a section in `CONTRIBUTING.md` about code style and tools.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -29,18 +29,23 @@ deps=
...
 
 [flake8]
 ignore=E731,W503,W504
 exclude=
+    .git,
     .tox,
-    __pycache__
-
+    __pycache__,
+    node_modules,
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply tox.ini]: Done

[the remark this answers is not in the document]
Comment (Scotchester) [reply tox.ini]: Is it? Wouldn't it prevent flake8 from going through the whole folder looking for `.py` files?

[the remark this answers is not in the document]
Comment (willbarton) [reply tox.ini]: There's no reason for there to ever be a `node_modules` directory in this tree, I don't think.

[the remark this answers is not in the document]
Comment (Scotchester) [reply tox.ini]: Ah, good point.

**Pull request merged**